ogr/ogrsf_frmts/gpkg/ogrgeopackagedatasource.cpp

```diff
@@ -1873,6 +1879,82 @@ void GDALGeoPackageDataset::FixupWrongMedataReferenceColumnNameUpdate()
     }
 }
 
+
+/************************************************************************/
+/*             LoadRelations()                                          */
+/************************************************************************/
+
+void GDALGeoPackageDataset::LoadRelations()
+{
+    m_osMapRelationships.clear();
+
+    auto oResultTable = SQLQuery(hDB, "SELECT base_table_name, base_primary_column, "
+                                 "related_table_name, related_primary_column, relation_name, "
+                                 "mapping_table_name FROM gpkgext_relations" );
+    if ( oResultTable && oResultTable->RowCount() > 0 )
+    {
+        for(int i=0; i<oResultTable->RowCount();i++)
+        {
+            const char* pszBaseTableName = oResultTable->GetValue(0, i);
```

Comment:
for robustness, you should assume that the return value of all those GetValue() might be nullptr (corrupted DB, etc.), and thus test for it before dereferencing/passing it to other functions